data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/DataPrepperServer.java

```diff
@@ -40,12 +48,24 @@ public DataPrepperServer(final DataPrepper dataPrepper) {
             throw new RuntimeException("Failed to create server", e);
         }
 
-        server.createContext("/metrics/prometheus", new PrometheusMetricsHandler());
-        server.createContext("/metrics/sys", new PrometheusMetricsHandler(DataPrepper.getSysJVMMeterRegistry()));
+        if (dataPrepperConfiguration.isMeterRegistryTypeConfigured(MeterRegistryType.Prometheus)) {
```

Comment:
Feels kind of off to have an explicit check here, but should be fine unless the types of meters and required server actions grows. Probably too much effort for little gain.